src/Microsoft.VisualStudio.ProjectSystem.Managed/MsBuildUtilities.cs

```diff
@@ -0,0 +1,172 @@
+
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    /// <summary>
+    /// Utitlies class to manipulate MsBuild projects.
+    /// </summary>
+    internal abstract class MsBuildUtilities
+    {
+        /// <summary>
+        /// Gets a project property.
+        /// </summary>
+        /// <param name="project">Xml representation of the MsBuild project.</param>
+        /// <param name="propertyName">Name of the property.</param>
+        /// <returns>Requested project property. Null if the property is not present.</returns>
+        public static ProjectPropertyElement GetProperty(ProjectRootElement project, string propertyName)
+        {
+            Requires.NotNull(project, "project");
+
+            return project.Properties
+                .Where(p => string.Equals(p.Name, propertyName, StringComparison.OrdinalIgnoreCase))
+                .FirstOrDefault();
+        }
+
+        /// <summary>
+        /// Gets the individual values of a delimited property.
+        /// </summary>
+        /// <param name="propertyValue">Value of the property to evaluate.</param>
+        /// <param name="delimiter">Character used to delimit the property values.</param>
+        /// <returns>Collection of individual values in the property.</returns>
+        public static ImmutableArray<string> GetPropertyValues(string propertyValue, char delimiter = ';')
+        {
+            var values = propertyValue.Split(delimiter).Select(f => f.Trim());
+
+            // We need to ensure that we return values in the specified order.
+            var valuesBuilder = ImmutableArray.CreateBuilder<string>();
+            foreach (var value in values)
+            {
+                if (!string.IsNullOrEmpty(value))
+                {
+                    valuesBuilder.Add(value);
+                }
+            }
+
+            return valuesBuilder.Distinct(StringComparer.OrdinalIgnoreCase).ToImmutableArray();
+        }
+
+        /// <summary>
+        /// Appends a value to a delimited property. If the property does not exist it will be added.
+        /// </summary>
+        /// <param name="project">Xml representation of the MsBuild project.</param>
+        /// <param name="evaluatedPropertyValue">Original evaluated value of the property.</param>
+        /// <param name="propertyName">Property name.</param>
+        /// <param name="propertyValue">Value to add to the property.</param>
+        /// <param name="delimiter">Character used to delimit the property values.</param>
+        public static void AppendPropertyValue(ProjectRootElement project, string evaluatedPropertyValue, string propertyName, string propertyValue, char delimiter = ';')
+        {
+            Requires.NotNull(project, nameof(project));
+            Requires.NotNull(evaluatedPropertyValue, nameof(evaluatedPropertyValue));
+            Requires.NotNullOrEmpty(propertyName, nameof(propertyName));
+
+            var property = GetOrAddProperty(project, propertyName);
```

Comment:
use `ProjectPropertyElement` instead of `var`? Reduces the confusion to think it returns a string